hyperspy/axes.py

```diff
@@ -592,21 +592,27 @@ def convert_to_uniform_axis(self):
             scale_err = max(self.axis[1:] - self.axis[:-1]) - scale
             _logger.warning('The maximum scale error is {}.'.format(scale_err))
         self.__class__ = UniformDataAxis
+        d["_type"] = 'UniformDataAxis'
```

Comment:
I don't think this is necessary - but I haven't check, so I may be wrong! ;)

Reply:
It is necessary only, if `_type` is an attribute, as otherwise it would keep the old `_type`. But if we put the `_type` in the export only, then it will be obsolete here and at a few other places, of course.

Reply:
> It is necessary only, if `_type` is an attribute, as otherwise it would keep the old `_type`.

It should get overwritten in the constructor, this is why I suspect it should not be necessary.

Reply:
In the current implementation, the constructor checks for consistency between the automatically chosen type and the passed parameter. Had been suggested by @francisco-dlp when proposing the **kwargs solution. But if `_type` is not kept as alias-attribute, it will be obsolete here.

Reply:
No, it is not obsolete, as long as `_type` is part of the dictionary and we have the consistency check.